### DISCUSSION TITLE
Does configuring SmtpSettings and other settings via appsettings.json still work?

### DISCUSSION THREAD

**Discussion by scleaver** [OrchardCMS/OrchardCore#10176] [Q&A]
Hi, according to the documentation here: https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/#ioptions-configuration

I should be able to configure the smtpsettings in the appsettings.json like so but this doesn't seem to work.

```
{
  "OrchardCore": {
    "DatabaseProvider": "SqlConnection",
    "SmtpSettings": {
      "Password":  "password"
    }
  }
}
```
Logic dictates that I should also then be able to set the password in App Service configuration like `OrchardCore__SmtpSettings__Password` - this doesn't seem to work either.

Outcome I am looking for - I want to default these settings for all tenants.

**Comment by scleaver** [comment]
Never mind it looks like I am misinterpreting the documentation based on https://github.com/OrchardCMS/OrchardCore/issues/6264

The documentation does make you think though that you can do this via appsettings.json but it appears you can not.

**Comment by deanmarcussen** [comment]
No worries, you can include them in appsettings, but you'd have to bind
them (I argued with Zoltan about including this in the docs a bit, because
it confuses the hell out of people they way he wrote it)

What you want to do is use PostConfigureOptions and bind the options value
from the app settings you made.

so inject IShellConfiguration and then run
PostConfigureOptions<SmtpSettings>(_shellConfiguration[SmtpSettings])
(roughly)

or pass a lambda PostConfigure<SmtpSettings>( o =>
{
 o.Password = _shellConfiguration.GetSection("SmtpSettings:Password")
}

Generally requires debugging, I can never remember the right accessor for
GetSection

D

**Reply by scleaver** [reply]
Thanks Dean that's really helpful.
